packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1076,9 +1076,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (_hasFocus) {
       _openInputConnection();
     } else {
-      final List<FocusScopeNode> ancestorScopes = FocusScope.ancestorsOf(context);
-      for (int i = ancestorScopes.length - 1; i >= 1; i -= 1)
-        ancestorScopes[i].setFirstFocus(ancestorScopes[i - 1]);
       FocusScope.of(context).requestFocus(widget.focusNode);
```

Comment:
oh wow, yeah, moving this out of clients is definitely a good move.

Reply:
Yeah, deleting that loop felt like a validation of the design.